### PR TITLE
Add days to fulfillment to product

### PR DESCRIPTION
**Pull request by PiotrZaranek**

### CODE REVIEW

File: be/src/Infrastructure/Catalog/Foundation.Catalog/Infrastructure/Products/Entities/Product.cs [hunk 1]
```diff
@@ -31,6 +31,8 @@ public class Product : Entity
 
         public string Ean { get; set; }
 
+        public int DaysToFulfilment { get; set; }
```
Comment (dawiddworak88) [inline be/src/Infrastructure/Catalog/Foundation.Catalog/Infrastructure/Products/Entities/Product.cs]: Hi @PiotrZaranek, it's fulfillment with a double 'l' inside. Let's rename it to FulfillmentTime where instead of days the amount of seconds is saved to the database and converted to days on the frontend, please

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]

File: be/src/Infrastructure/Catalog/Foundation.Catalog/Infrastructure/Products/Entities/Product.cs [hunk 2]
```diff
@@ -31,6 +31,8 @@ public class Product : Entity
 
         public string Ean { get; set; }
 
+        public int DaysToFulfillment { get; set; }
```
Comment (SebaKas21) [inline be/src/Infrastructure/Catalog/Foundation.Catalog/Infrastructure/Products/Entities/Product.cs]: **suggestion:** It should be FulfillmentTime according to @dawiddworak88  suggestion

File: fe/projects/Seller.Portal/src/areas/Products/components/ProductForm/ProductForm.js [hunk 3]
```diff
@@ -133,6 +134,7 @@ function ProductForm(props) {
             files,
             isNew,
             ean,
+            daysToFulfillment,
```
Comment (SebaKas21) [inline fe/projects/Seller.Portal/src/areas/Products/components/ProductForm/ProductForm.js]: **suggestion:** Should be fulfillmentTime instead of daysToFulfillment

File: fe/projects/Seller.Portal/src/areas/Products/components/ProductForm/ProductForm.js [hunk 4]
```diff
@@ -283,6 +285,10 @@ function ProductForm(props) {
                             <TextField id="name" name="name" label={props.nameLabel} fullWidth={true} variant="standard"
                                 value={name} onChange={handleOnChange} helperText={dirty.name ? errors.name : ""} error={(errors.name.length > 0) && dirty.name} />
                         </div>
+                        <div className="field">
+                            <TextField type="number" id="daysToFulfillment" name="daysToFulfillment" label={props.daysToFulfillmentLabel} fullWidth={true} variant="standard"
+                            value={daysToFulfillment} onChange={handleOnChange} />
```
Comment (SebaKas21) [inline fe/projects/Seller.Portal/src/areas/Products/components/ProductForm/ProductForm.js]: **suggestion:** Set to fulfillmentTime and convert to days

Example:
```js
moment.utc(1000 * fulfillmentTime ).format("DD");
```

**Review by SebaKas21 — CHANGES_REQUESTED** [review]

File: fe/projects/Seller.Portal/src/areas/Products/components/ProductForm/ProductForm.js [hunk 5]
```diff
@@ -47,7 +48,8 @@ function ProductForm(props) {
         uiSchema: { value: props.uiSchema ? JSON.parse(props.uiSchema) : {} },
         formData: { value: props.formData ? JSON.parse(props.formData) : {} },
         isPublished: { value: props.isPublished ? props.isPublished : false },
-        ean: { value: props.ean ? props.ean : "" }
+        ean: { value: props.ean ? props.ean : "" },
+        fulfillmentTime: { value: Number(moment.utc(1000 * props.fulfillmentTime).format("DD") - 1)}
```
Comment (SebaKas21) [inline fe/projects/Seller.Portal/src/areas/Products/components/ProductForm/ProductForm.js]: **suggestion** Check if fulfillmentTime have value

File: fe/projects/Seller.Portal/src/areas/Products/components/ProductForm/ProductForm.js [hunk 6]
```diff
@@ -133,10 +135,11 @@ function ProductForm(props) {
             files,
             isNew,
             ean,
+            fulfillmentTime: moment.duration(fulfillmentTime, 'days').asSeconds(),
```
Comment (SebaKas21) [inline fe/projects/Seller.Portal/src/areas/Products/components/ProductForm/ProductForm.js]: **suggestion** also here check if fulfillmentTime has value

File: be/src/Infrastructure/Catalog/Foundation.Catalog/Infrastructure/Products/Entities/Product.cs [hunk 7]
```diff
@@ -31,6 +31,8 @@ public class Product : Entity
 
         public string Ean { get; set; }
 
+        public int FulfillmentTime { get; set; }
```
Comment (SebaKas21) [inline be/src/Infrastructure/Catalog/Foundation.Catalog/Infrastructure/Products/Entities/Product.cs]: **suggestion** It should be nullable if it isn't required. If you change please generate new migration

**Review by SebaKas21 — CHANGES_REQUESTED** [review]
Hi @PiotrZaranek 

PLease also look on this error

```sql
ALTER TABLE [CategoryImages] ADD CONSTRAINT [FK_CategoryImages_Categories_CategoryId] FOREIGN KEY ([CategoryId]) REFERENCES [Categories] ([Id]) ON DELETE CASCADE;
Unhandled exception. Microsoft.Data.SqlClient.SqlException (0x80131904): The ALTER TABLE statement conflicted with the FOREIGN KEY constraint "FK_CategoryImages_Categories_CategoryId". The conflict occurred in database "CatalogDb", table "dbo.Categories", column 'Id'.
```

File: be/src/Project/Services/Catalog/Catalog.Api/Infrastructure/Migrations/CatalogContextModelSnapshot.cs [hunk 8]
```diff
@@ -536,6 +546,15 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                     b.ToTable("ProductVideos");
                 });
 
+            modelBuilder.Entity("Foundation.Catalog.Infrastructure.Categories.Entites.CategoryImage", b =>
```
Comment (dawiddworak88) [inline be/src/Project/Services/Catalog/Catalog.Api/Infrastructure/Migrations/CatalogContextModelSnapshot.cs:549]: **suggestion:** let's move category image related changes to a separate PR

File: be/src/Infrastructure/Localization/Foundation.Localization/Resources/ProductResources.en.resx [hunk 9]
```diff
@@ -159,6 +159,9 @@
   <data name="EnterSkuText" xml:space="preserve">
     <value>Enter SKU</value>
   </data>
+  <data name="FulfillmentTimeLabel" xml:space="preserve">
+    <value>Days To Fulfilment:</value>
```
Comment (dawiddworak88) [inline be/src/Infrastructure/Localization/Foundation.Localization/Resources/ProductResources.en.resx]: suggestion: Fulfillment with double 'l inside

File: be/src/Project/Web/Seller/Seller.Web/Areas/Products/ApiRequestModels/SaveProductRequestModel.cs [hunk 10]
```diff
@@ -17,6 +17,7 @@ public class SaveProductRequestModel : RequestModelBase
         public IEnumerable<SaveFileRequestModel> Files { get; set; }
         public bool IsNew { get; set; }
         public string Ean { get; set; }
+        public int FulfillmentTime { get; set; }
```
Comment (dawiddworak88) [inline be/src/Project/Web/Seller/Seller.Web/Areas/Products/ApiRequestModels/SaveProductRequestModel.cs]: **suggestion:** Let's make it nullable and optional on the product card. It should put null into the database if no fulfillment time has been set

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]

File: be/src/Project/Web/Seller/Seller.Web/Areas/Products/ApiRequestModels/SaveProductRequestModel.cs (hunk 10 above)
Comment (SebaKas21) [inline be/src/Project/Web/Seller/Seller.Web/Areas/Products/ApiRequestModels/SaveProductRequestModel.cs]: **suggestion** Please also set int as nullable

**Review by SebaKas21 — CHANGES_REQUESTED** [review]

**Review by SebaKas21 — APPROVED** [review]

**Review by dawiddworak88 — APPROVED** [review]
Thank you @PiotrZaranek, thank you @SebaKas21
